### DISCUSSION TITLE
Health check before scheduled backup

### DISCUSSION THREAD

**Discussion by adrian-salazar-klarrio** [vmware-tanzu/velero#6641] [Community support Q&A]
Hello, we currently have velero running with an hourly scheduled backup, but sometimes the cluster is not fully stable and the backup is done in this status. 

Is there a way to do a custom check before the schedule is triggered? The hooks provided are only applied to pods. I see this was already asked here without answer: https://github.com/vmware-tanzu/velero/discussions/3603

In case it is not possible, any ideas? I've runned out of... are there any plans to include this feature?

**Comment by draghuram** [comment]
As you noticed, there are no hooks in Velero that run before a backup starts. There are multiple ways of achieving what you want to do though they may not exactly be "clean" solutions:

- Run your own CronJob that will run few minutes before the Velero schedule. If you don't want backup to proceed, you can "pause" the schedule. If you do want to let it run,"unpause" the schedule if needed.
- Run your own CronJob as above but instead of using schedule, create your own "Backup" CR every time you want backup to run.

**Comment by sseago** [comment]
This feature design might do what you want. It would implement pre/post backup (and restore) action types. It would be worth reading through the design to see whether it might meet your needs here. One thing to look for -- I don't remember whether that design accounts for a response on the pre- actions that says "don't run the backup at all", but an error return from the action might accomplish the same thing.
https://github.com/vmware-tanzu/velero/pull/6466
